### PR TITLE
Refactor mass2mz and mz2mass as detailed in issue #29

### PR DESCRIPTION
**Pull request by jorainer**

### CODE REVIEW

File: R/adducts.R [hunk 1]
```diff
@@ -7,10 +7,13 @@
...
-#' @param adduct `character` specifying the name of the adduct;
-#'     supported values are returned by [adductNames()]
+#' @param adduct `character` specifying the name(s) of the adduct(s) for which
+#'     the m/z should be calculated; supported values are returned by
+#'     [adductNames()]
 #'
-#' @return `numeric` representing the calculated m/z value.
+#' @return numeric `matrix` with same number of rows than elements in `x` and
+#'     number of columns being equal to the length of `adduct`. Each column
+#'     thus represents the m/z of `x` for each defined `adduct`.
```
Comment (michaelwitting) [inline R/adducts.R:16]: Does the matrix have column names?

Comment (jorainer) [reply R/adducts.R:16]: Yes it does. The column names are the provided adduct names. I will add that to the documentation.

**Comment by jorainer** [comment]
Note @michaelwitting , I would still like to make some changes to the PR. I'll ask you for review again once I'm done with that.

**Comment by jorainer** [comment]
OK, included the last changes to support also custom adduct definition for `mass2mz` and `mz2mass` in form of a `data.frame`.

File: R/adducts.R [hunk 2]
```diff
@@ -39,12 +46,15 @@
 #'
 #' ## Calculate m/z of multiple adducts from neutral mass
 #' mass2mz(exact_mass, adduct = adductNames())
+#'
+#' ## Provide a custom adduct definition.
+#' adds <- data.frame(mass_add = c(1, 2, 3), mass_multi = c(1, 2, 0.5))
+#' rownames(adds) <- c("a", "b", "c")
+#' mass2mz(c(100, 200), adds)
 mass2mz <- function(x, adduct = "[M+H]+") {
-    idx <- match(adduct, names(.ADDUCTS_ADD))
```
Comment (michaelwitting) [inline R/adducts.R:39]: In case of custom adducts this will fail, or?

Comment (jorainer) [reply R/adducts.R:39]: Sure, but I removed these lines (that's why they are red) and replaced it with the (green) lines below.

Comment (michaelwitting) [reply R/adducts.R:39]: Man... too late... of course! ;-)

**Pull request merged**